Sources/EnhancedMediaPlayer/SettingsMenu.swift

```diff
@@ -1,18 +1,33 @@
 // SettingsMenu.swift
+
 import SwiftUI
 
 public struct SettingsMenu: View {
-    public init() { }
+    @ObservedObject private var preferences: SettingsPreferences
+
+    public init(preferences: SettingsPreferences) {
+        self.preferences = preferences
+    }
 
     public var body: some View {
         VStack(spacing: Constants.stackSpacing) {
             renderSettingsItem()
-            renderSettingsItem()
-            renderSettingsItem()
+            renderPlaybackSpeedMenu()
         }
         .frame(maxHeight: .infinity, alignment: .topLeading)
         .padding(Constants.stackPadding)
     }
+    
+    @ViewBuilder func speedButton(speed: Float) -> some View {
```

Comment:
This need to be below `renderPlaybackSpeedMenu`
Also it would be good to rename this to `renderSpeedMenuOption`

Reply:
Also how about using `speed: PlaybackSpeedValues`? this way you can constrain the possible values of speed